05/app.js

```diff
@@ -3,41 +3,61 @@ const msgList = document.querySelector('.messages');
 const submit = document.querySelector('input[type="submit"]')
 formEl.setAttribute('novalidate', true)
 
-submit.addEventListener('submit', function(e) {
-    e.preventDefault();
+formEl.addEventListener('submit', function (e) {
     validateInfo(e);
 })
 
 function validateInfo(e) {
-    e.preventDefault()
-
+    e.preventDefault();
+    msgList.innerHTML = '';
     const errors = [];
-    const firstName = e.target.elements.firstName.value.trim();
-    const lastName = e.target.elements.lastName.value.trim();
-    const street = e.target.elements.street.value.trim();
-    const houseNumber = e.target.elements.houseNumber.value.trim();
-    const flatNumber = e.target.elements.flatNumber.value.trim();
-    const zipCode = e.target.elements.zip.value.trim();
-    const city = e.target.elements.city.value.trim();
-    const region = e.target.elements.voivodeship.value.trim();
-
-    if(firstName === '') {
-        addError(errors, firstName);
-    }
-}
 
+    const firstNameEl = e.target.elements.firstName
+    const lastNameEl = e.target.elements.lastName
+    const streetEl = e.target.elements.street
+    const houseNumberEl = e.target.elements.houseNumber
+    const zipEl = e.target.elements.zip
+    const cityEl = e.target.elements.city
+    const regionEl = e.target.elements.voivodeship
 
-function checkNames(name) {
-    const regName = /^[a-zA-Z]+ [a-zA-Z]+$/;
-    if(!regName.test(name)) {
-        addError(errorsArray, name)
-    } 
-}
+    const firstNameValue = firstNameEl.value.trim();
+    const lastNameValue = lastNameEl.value.trim();
+    const streetValue = streetEl.value.trim();
+    const houseNumberValue = houseNumberEl.value.trim();
+    const zipCodeValue = zipEl.value.trim();
+    const cityValue = cityEl.value.trim();
+    const regionValue = regionEl.value.trim();
+    const zipPattern = /^[0-9]{2}-[0-9]{3}$/
 
-function addError(array, inputEl) {
-    const labelContent = inputEl.parentElement.textContent;
-    const errorMsg = `${labelContent} is required`;
-    array.push(errorMsg)
+    if (firstNameValue === '') {
```

Comment:
Duża ilość if-ów jest raczej źle oceniana - niepotrzebnie komplikuje rozubdowę kodu.
Dodanie 10 pól formularza to kolejne 10 if-ów. Dobrze by było, aby walidacja była elastyczna.

Traktuj to jako zadanie dodatkowe. Jeśli chcesz spróbować zmienić podejście to możesz wzorować się na tym rozwiązaniu: https://devmentor.pl/b/walidacja-formularza-w-javascript